src/Microsoft.AspNetCore.Blazor/RenderFragmentExtensions.cs

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.AspNetCore.Blazor
+{
+    /// <summary>
+    /// Contains extension methods for <see cref="RenderFragment"/> and <see cref="RenderFragment{T}"/>.
+    /// </summary>
+    public static class RenderFragmentExtensions
+    {
+        /// <summary>
+        /// Binds a <see cref="RenderFragment{T}" /> and a value of type <typeparamref name="T"/> to a
+        /// <see cref="RenderFragment"/> so that it can be used by the rendering system from Razor code.
+        /// </summary>
+        /// <typeparam name="T">The type of the value used by the <paramref name="fragment"/>.</typeparam>
+        /// <param name="fragment">A <see cref="RenderFragment{T}"/>, usually produced by a Razor template.</param>
+        /// <param name="value">The value of type <typeparamref name="T"/>.</param>
+        /// <returns>A <see cref="RenderFragment"/>.</returns>
+        public static RenderFragment Bind<T>(this RenderFragment<T> fragment, T value)
+        {
+            return (b) => fragment(b, value);
+        }
```

Comment:
@SteveSandersonMS  - interested in your suggestions here. Basically you have no way to consume a `RenderFragment<T>` from 'loose' Razor code. Example:

```
@{ RenderFragment<Person> p = @<div>...</div>; }
...
<div>
   @p     @* doesn't work the render tree builder doesn't understand RenderFragment<Person> *@
</div>
<div>
   @p(new Person(...))    @* doesn't work, wrong number of args *@
</div>
<div>
   @p(???, new Person(...))    @* doesn't work, I don't have a builder to pass in *@
</div>
```

Also, credit to @uazo - I was inspired by some of the things he tried.

Reply:
The approach looks fine and mirrors how we handle some other cases where we have a `RenderFragment` (for example, how we lift child content into a lambda that takes a `RenderTreeBuilder`). I know it costs a heap allocation for the delegate but we have no better alternative.

My personal preference would be to avoid the name `Bind` just because it's so overloaded. Maybe `WithValue`?

Reply:
sure